tests/runner/src/env.rs

```diff
@@ -183,6 +183,13 @@ impl Env {
             _ => panic!("Unexpected subcommand: {subcommand}"),
         };
 
+        if util::check_port(check_ip_addr.parse().unwrap(), Duration::from_secs(1)).await {
```

Comment:
Maybe we can refactor this `check_port` to receive an optional duration. For this scenario we only need to check the port once.

Reply:
ok,i will try to refactor it.😄

Reply:
The two checks correspond to two different logics. The first check judges whether the port is occupied, and the second check judges whether the process is up or not(i dont know whether i understand it enough).i am confused about how to mix these two logics into one(as you said "check the port once")
and i found this:
```rust
let mut process = Command::new("./greptime")
            .current_dir(util::get_binary_dir("debug"))
            .args(args)
            .stdout(log_file)
            .spawn()
            .unwrap_or_else(|_| panic!("Failed to start the DB with subcommand {subcommand}"));
```
if up the process fail,this is already a panic,Is the second `check_port` redundant?

Reply:
@waynexia thx for trust🌹,i will have a try.

Reply:
Going back to the original modification, I found that both check_ports are necessary😭😭😭.
> Maybe we can refactor this `check_port` to receive an optional duration. For this scenario we only need to check the port once.

Could you explain this refactor in more detail? I may not have fully understood.sry😭

Reply:
Before starting the server, we only need to check the port "once" without delay. Thus we don't need `timeout`. And after the server is started, it might take some time to initialize. So we check the port repeatedly for a `timeout` duration.

But this is an optimization anyway. It should not affect correctness.

Reply:
I understand you now,i will try to refractor it,thx!